portal-impl/src/com/liferay/portal/verify/VerifyProperties.java

```diff
@@ -41,9 +42,17 @@ public class VerifyProperties extends VerifyProcess {
 	protected void doVerify() throws Exception {
 		verifySystemProperties();
 
-		verifyPortalProperties();
+		List<String> keys = verifyPortalProperties();
 
 		verifyDocumentLibrary();
+
+		if (!keys.isEmpty()) {
+			System.out.println(
+				"Stopping the server due to incorrect use of migrated Portal " +
+					"properties " + keys);
+
+			System.exit(0);
```

Comment:
@W2jx There is a convention in both Windows and Unix that a process returns `0` to indicate that there's no error during the execution. So here we should not use `0` as the exit code.

In Windows, it's the application's decision for what non-zero exit codes mean;
In Linux, there's a reserved exit code list with special meanings: https://tldp.org/LDP/abs/html/exitcodes.html.

Based on this, `System.exit(1)` should be enough for us for now.

Reply:
I'm sorry, I misunderstood them, and I will fix it.